### ISSUE TITLE
Incompatibility with .gitignore pattern

### ISSUE THREAD

**Issue by cprecioso** [martinvonz/jj#3225]
## Description

`.gitignore` admits the following pattern:

```
foo/*
!foo/bar
```

It basically ignores everything inside of `foo` instead of the `bar` path.
Git interprets this correctly, jj ignores the file.

This is used, for example, in `yarn`'s (quite popular JS package manager) template for a new package: https://github.com/yarnpkg/berry/blob/2aa53d144c2e0fed8792032ce8655a34116a15f7/packages/plugin-init/sources/commands/init.ts#L175-L181

## Steps to Reproduce the Problem

1. `jj git init --colocate`

2. Create `.gitignore` with
	```
	foo/*
	!foo/bar
	```
	
4. Create the `foo/bar/file` file
5. `git add . && git status` shows the newly added file
6. `jj status` doesn't

## Expected Behavior

jj tracks the `foo/bar/file`

## Actual Behavior

jj ignores the `foo/bar/file`

## Specifications

- Platform: macOS 14.3.1
- Version: 0.14.0

**Comment by yuja** [comment]
Probably a duplicate of #3031, which appears to be fixed in the main branch.

**Comment by cprecioso** [comment]
Yes, just tried it installing from main, and the problem seems solved.
Thanks!
